pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -127,3 +128,80 @@ def close(self) -> None:
         if not self.skip_check_coredump:
             self._check_coredump()
         super().close()
+
+    case_regex = re.compile(r'\((\d+)\)\s\"(.+)\"\s(\[.+\])+')
+    subcase_regex = re.compile(r'\t\((\d+)\)\s\"(.+)\"')
+
+    def get_test_menu(self) -> List:
+        """
+        Get test case list from test menu via UART print.
+
+        This function should be called after `Press ENTER to see the list of tests` prompt is shown.
+
+        Returns:
+            A `list` of avaliable test cases, in `dict`. Each test case's `dict` has following keys:
```

Comment:
Could we use an instance of NamedTuple or a class instead of a `dict`? It's a bit more maintainable than a dictionary, plus it's better for documentation and for autocompletion hints in IDEs.